src/coreclr/src/vm/dllimportcallback.cpp

```diff
@@ -1194,33 +1209,20 @@ VOID UMThunkMarshInfo::RunTimeInit()
 
     if (pFinalILStub == NULL)
     {
-        if (pMD != NULL && !pMD->IsEEImpl() &&
-            !NDirect::MarshalingRequired(pMD, GetSignature().GetRawSig(), GetModule()))
-        {
-            // Call the method directly in no-delegate case if possible. This is important to avoid JITing
-            // for stubs created via code:ICLRRuntimeHost2::CreateDelegate during coreclr startup.
-            pFinalILStub = pMD->GetMultiCallableAddrOfCode();
-        }
```

Comment:
Can you unpack this issue for me? If the `MethodDesc` exists, but it wasn't implemented by the runtime and it is a blittable signature, just return the stub? Is this no longer interesting or am I missing why this is no longer a scenario.

Reply:
We always need to generate the IL stub with the GC transition now, even where there is not marshaling required. I could have kept this for non-Windows x86 but it did not seem worth the complexity. The non-Windows x86 path is pretty inefficient in general.

The comment about the CreateDelegate case is from Silverlight days. It is not that important now. That makes me think: We should allow CreateDelegate to return direct pointer to NativeCallable methods: https://github.com/dotnet/runtime/issues/34266